### PR TITLE
Add Producers Tutorial

### PR DESCRIPTION
**Pull request by DoubleA-MG**

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -65,6 +65,7 @@ nav:
           - Perk Systems: tutorials/perks_tutorial.md
           - Perk Systems Part 2: tutorials/perks_tutorial_part2.md
           - Persistent Storage: tutorials/persistent_storage_tutorial.md
+          - Producers: tutorials/producers_tutorial.md
```
Comment (CommanderFoo) [inline mkdocs.yml]: Add ✨ since it's a new item.

File: src/tutorials/producers_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,1100 @@
...
+9. Use the **Hammer** item to break down and collect placed items.
+
+<div class="mt-video" style="width:100%">
+    <video autoplay muted playsinline controls loop class="center" style="width:100%">
+        <source src="/img/ProducersTutorial/Producer_FruitExample.mp4" type="video/mp4" />
+    </video>
+</div>
+
+!!! warning
+    While in Preview Mode, players can press ++F1++ to **reset the Player's Data**. This is useful for testing new items or features in the game. However, this should be removed before publishing a game.
```
Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md]: Maybe include a little tip on where to disable it. I took a look at the framework myself just now, and could not find the setting to turn it off.

File: src/tutorials/producers_tutorial.md [hunk 3]
```diff
@@ -0,0 +1,1100 @@
...
+12. Search the Hierarchy for the `Alien Egg` script and open the Properties window.
+13. Set the **IconAsset** property to **Alien Egg (Inventory Icon)**.
+
+![!Alien Egg IconAsset](../img/ProducersTutorial/Producer_AlienEgg_IconAsset.png){: .center loading="lazy" }
+
+#### Visualize the Icon
+
+As long as the **Alien Egg (Inventory Icon)** matched the red fruit Icon Asset template, then it will most likely produce a clear inventory icon. However, it is good to know how the **Icon Generator** is placing the template in front of the capturing camera in case adjustments are needed.
+
+1. In the Hierarchy, expand the **UI** folder and right click the **Icon Generator** and select **Desinstance This Object**.
```
Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md]: Typo: Desinstance

Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md]: The Icon Generator in the UI folder isn't a template.

File: src/tutorials/producers_tutorial.md [hunk 4]
```diff
@@ -0,0 +1,1100 @@
...
+
+### Add to Starting Items
+
+The Burnt Egg item is now ready to be tested so it will need to be added as a starting item in the **Hotbar Inventory**.
+
+1. In the Hierarchy, search for the `Starting Items` script under the **Hotbar Settings** group.
+2. Right click the **Alien Egg** script and select **Duplicate**.
+3. Rename the duplicated script to `Burnt Egg`.
+4. Select the **Burnt Egg** script and open the Properties window.
+5. Set the ItemId property to Burnt Egg.
```
Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md]:
```suggestion
5. Set the **ItemId** property to `Burnt Egg`.
```

File: src/tutorials/producers_tutorial.md [hunk 5]
```diff
@@ -0,0 +1,1100 @@
...
+6. Set the SlotIndex property to 3.
```
Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md]:
```suggestion
6. Set the **SlotIndex** property to `3`.
```

File: src/tutorials/producers_tutorial.md [hunk 6]
```diff
@@ -0,0 +1,1100 @@
...
+
+### Create the Item Data
+
+In the Hierarchy, open the **Global Database** folder and expand the **Items** script. Right click the **Alien Egg** script and select **Duplicate**. Rename the duplicated script to `Alien Raptor`.
+
+![!Raptor Item](../img/ProducersTutorial/Producer_Raptor_Item.png){: .center loading="lazy" }
+
+### Set the Item Properties
+
+The Alien Raptor is not be equippable so an equipment template is not needed.
```
Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md]: "is not be equippable"

File: src/tutorials/producers_tutorial.md [hunk 7]
```diff
@@ -0,0 +1,1100 @@
...
+#### Create a Collision Cube
+
+The **Collision Cube** needs to activate the **Can Overlap Triggers** property to avoid **overlapping Placeables**.
+
+1. Search the **Project Content** window for the `Incubator (Drop Icon)` template.
+2. Drag and drop the **Incubator (Drop Icon)** template into the scene.
+3. Right click the template and select **Deinstance This Object**.
+4. Search **Core Content** for a `Cube` mesh.
+5. Drag and drop the **Cube** into the **Geo** group.
+6. Transform the **Cube** match the dimensions of the **Incubator**.
```
Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md]: "Transform the **Cube** match the"

File: src/tutorials/producers_tutorial.md [hunk 8]
```diff
@@ -0,0 +1,1100 @@
...
+The **Trigger Template** will use triggers to check if a Placeable is going to overlap other triggers when being placed.
+
+#### Create the Trigger
+
+1. Search the **Project Content** window for the `Incubator (Drop Icon)` template.
+2. Drag and drop the **Incubator (Drop Icon)** template into the scene.
+3. Right click the template and select **Deinstance This Object**.
+4. Search **Core Content** for a `Trigger`.
+5. Drag and drop the **Trigger** into the **Incubator (Drop Icon)** template.
+6. Transform the **Trigger** match the dimensions of the **Incubator Model**.
```
Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md]: "Transform the **Trigger** match the"

File: src/tutorials/producers_tutorial.md [hunk 9]
```diff
@@ -0,0 +1,1100 @@
...
+
+### Create the Client Template
+
+The **Terminal (Drop Icon)** was scaled down to look more like a pickup, but the **Placeable Client Template** needs to be larger for the in-game size.
+
+1. Search **Project Content** for the `Terminal (Drop Icon)` template.
+2. Drag and drop the **Terminal (Drop Icon)** template into the scene.
+3. Right click the template and select **Deinstance this Object**.
+4. Scale up the **Terminal Model** to the in-game size.
+5. Right click the **Terminal (Drop Icon)** template and select **Create New Template From Thi**s.
```
Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md]:
```suggestion
5. Right click the **Terminal (Drop Icon)** template and select **Create New Template From This**.
```

File: src/tutorials/producers_tutorial.md [hunk 10]
```diff
@@ -0,0 +1,1100 @@
...
+Preview the game and attempt to place the Terminal and Incubator objects on the ground.
+
+The following should work:
+
+- Placeables can only be placed on the Scifi area.
+- Incubator lights up green if within the Terminal radius.
+- Incubator lights up red if not within Terminal radius.
+- Placeables should be prevented from being placed on one another.
+- Players should have collision with placed objects.
+- Placeables should be saved across game plays.
```
Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md]: "saved across game plays" -> "saved across game sessions"

File: src/tutorials/producers_tutorial.md [hunk 11]
```diff
@@ -0,0 +1,1100 @@
...
+
+A **Producer** uses **phases** to control the produced drops and the visuals.
+
+These are the four phases being used for the **Incubating Egg**:
+
+- **Placed**: When a player places the egg on the incubator it enters this phase.
+- **Building**: If a placed egg is receiving the Electricity Buff, then it begins the building phase.
+- **Ready**: Once the building phase has gone on for a couple seconds, it is now ready to be harvested.
+- **Expired**: If not harvested when ready for a couple more seconds, then the egg expires and becomes burnt.
+
```
Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md:972]: Maybe switch the above list to a table with the headings "Phase" and "Description"?

File: src/tutorials/producers_tutorial.md [hunk 12]
```diff
@@ -0,0 +1,1100 @@
...
+The **Static Template** is an invisible cube that handles collision.
+
+#### Create the Collision Cube
+
+The **Collision Cube** will use the **Terminal (Client)** template as a size reference.
+
+1. Right click the **Terminal (Client)** template and select **Deinstance This Object**.
+2. Search **Core Content** for a `Cube` mesh.
+3. Drag and drop the **Cube** into the **Geo** group.
+4. Transform the **Cube** match the dimensions of the **Terminal**.
```
Comment (CommanderFoo) [inline src/tutorials/producers_tutorial.md]: "Transform the **Cube** match"

File: src/tutorials/producers_tutorial.md (hunk 3 above)
Comment (DoubleA-MG) [reply src/tutorials/producers_tutorial.md]: Are you sure? I just tested on a clean project and it seems to be a template
template
.

Comment (CommanderFoo) [reply src/tutorials/producers_tutorial.md]: For me (new project), it's not a template.

**Review by CommanderFoo — APPROVED** [review]

**Pull request merged**